base/channels.jl

```diff
@@ -129,8 +129,8 @@ function Channel(func::Function; ctype=Any, csize=0, taskref=nothing)
     isa(taskref, Ref{Task}) && (taskref[] = task)
     return chnl
 end
-function Channel{T}(f::Function, sz=0) where T
-    return Channel(f, csize=sz, ctype=T)
+function Channel{T}(f::Function, size) where T
+    return Channel(f, csize=size, ctype=T)
 end
 function Channel{T}(f::Function; csize=0, taskref=nothing) where T
     return Channel(f, csize=csize, ctype=T, taskref=taskref)
```

Comment:
@JeffBezanson While i'm here, I wonder if it was a mistake to keep the kwargs in this separate constructor.

Ideally, what i would have done is delete the `csize=0` and `ctype=` kwargs, and merge the constructors together so that we just had two types of constructors:
- `Channel{T=Any}(func::Function, size=0; taskref=nothing)`: Creates a task w/ the provided `func`
- `Channel{T=Any}(sz=0)`: Just creates a Channel

But since we can't delete existing params (no breaking changes), we had to just add _more_ options.

Do you think it would make sense to also add the optional `; taskref=nothing` argument to the `Channel{T}(f::Function, size)` constructor as well?